### PR TITLE
[WebSocket] Remove unnecessary and duplicate code of WebSocket implementation

### PR DESCRIPTION
**Pull request by irunika**

### CODE REVIEW

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketConnectorFuture.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    void setWSConnectorListener(WebSocketConnectorListener connectorListener);
+
+    /**
+     * Notify WebSocket connection initialization for the listener.
+     *
+     * @param initMessage {@link WebSocketInitMessage} to notify connection initialization.
+     * @throws WebSocketConnectorException if any error occurred during the notification.
+     */
+    void notifyWSListener(WebSocketInitMessage initMessage) throws WebSocketConnectorException;
```
Comment (riyafa) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketConnectorFuture.java]: Shall we rename to `notifyWebSocketListener`?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketConnectorFuture.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            throws WebSocketConnectorException;
+
+    /**
+     * Notify idle timeout of WebSocket connection.
+     *
+     * @param controlMessage Indicate WebSocket connection timeout.
+     * @throws WebSocketConnectorException if any error occurred during notification to the listener.
+     */
+    void notifyWSIdleTimeout(WebSocketControlMessage controlMessage) throws WebSocketConnectorException;
+}
```
Comment (riyafa) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketConnectorFuture.java]: Shall we change WS in all places to WebSocket in method names?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/DefaultWebSocketConnectorFuture.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+/**
+ * Default implementation of {@link WebSocketConnectorFuture}.
+ */
+public class DefaultWebSocketConnectorFuture implements WebSocketConnectorFuture {
+
+    private WebSocketConnectorListener wsConnectorListener;
+
+
+    @Override
+    public void setWSConnectorListener(WebSocketConnectorListener wsConnectorListener) {
```
Comment (riyafa) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/DefaultWebSocketConnectorFuture.java]: Here also WS->WebSocket

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/HttpConnectorFuture.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+/**
+ * Connector Future for HTTP events.
+ */
+public interface HttpConnectorFuture {
+
+    /**
+     * Set Connector listener for HTTP.
+     *
+     * @param connectorListener Connector listener for HTTP.
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/HttpConnectorFuture.java]: Shall we remove the period added for parameter docs? It is usually added only for the descriptions.

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketConnectorFuture.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+package org.wso2.transport.http.netty.contract.websocket;
+
+/**
+ * Connector Future for WebSocket events.
+ */
+public interface WebSocketConnectorFuture {
+
+    /**
+     *  Set Connector listener for WebSocket.
+     * @param connectorListener Connector listener for WebSocket.
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketConnectorFuture.java]: Leave a new line

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/DefaultWebSocketConnectorFuture.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        this.wsConnectorListener = wsConnectorListener;
+    }
+
+    @Override
+    public void notifyWSListener(WebSocketInitMessage initMessage) throws WebSocketConnectorException {
+        if (wsConnectorListener == null) {
+            throw new WebSocketConnectorException("WebSocket connector listener is not set");
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/DefaultWebSocketConnectorFuture.java]: Shall we introduce a String constant for "WebSocket connector listener is not set" ? Same has been used redundantly.

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/HttpConnectorFuture.java (hunk 4 above)
Comment (irunika) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/HttpConnectorFuture.java]: Done

**Pull request merged**